README.md
# osint
For the osint project in infosys3858


history.txt
  996  cat /etc/networking/resolv.conf
  997  cat /etc/network/interfaces
  998  sudo nano /etc/network/interfaces
  999  ifconfig
 1000  cat /etc/network/interfaces
 1001  sudo nano /etc/network/interfaces
 1002  sudo source /etc/network/interfaces
 1003  source /etc/network/interfaces
 1004  service networking restart
 1005  ifconfig
 1006  sudo nano /etc/network/interfaces
 1007  service networking restart
 1008  ifconfig
 1009  cat /etc/resolv.conf
 1010  sudo nano /etc/resolv.conf
 1011  shutdown
 1012  shutdown -s -t 00
 1013  shutdown now
 1014  ping google.com
 1015  ifconfig
 1016  ip addr
 1017  sudo apt-get install network-manager
 1018  update
 1019  c
 1020  nano /etc/network/interfaces
 1021  shutdow now
 1022  shutdown now
 1023  ifconfig
 1024  ip addr
 1025  cat /etc/network/interfaces
 1026  sudo nano .bashrc
 1027  source .bashrc
 1028  update
 1029  shutdown now
 1030  cd Desktop/Problem1/Shell_Scripts/
 1031  cat install.txt
 1032  history
 1033  history > history.html 
 1034  sudo history > history.html 
 1035  chmod 775 history.html 
 1036  history > history.html 
 1037  chmod 444 history.html 
 1038  ll
 1039  c
 1040  cd ..
 1041  mkdir Problem1
 1042  cd Problem1/
 1043  mkdir Shell_Scripts Group1 Group2
 1044  ls
 1045  ll
 1046  cd Shell_Scripts/
 1047  touch install.sh
 1048  ll
 1049  chmod +x install.sh 
 1050  ll
 1051  nano install.sh 
 1052  install.sh
 1053  ./install.sh
 1054  nano install.sh 
 1055  c
 1056  ./install.sh
 1057  nano install.sh 
 1058  c
 1059  ./install.sh
 1060  zenmap
 1061  sudo zenmap
 1062  sudo su
 1063  update
 1064  sudo nano .bashrc
 1065  c
 1066  cd Desktop/
 1067  cd Problem1/
 1068  cd Group1
 1069  ls
 1070  nano SampleText.txt
 1071  chmod 751 SampleText.txt 
 1072  nano anotherSampleText.txt
 1073  chmod 777 anotherSampleText.txt 
 1074  ls
 1075  ll
 1076  cd ..
 1077  cd Group2
 1078  cp /Desktop/Problem1/Group1/SampleText.txt
 1079  cp /Desktop/Problem1/Group1/SampleText.txt /Desktop/Problem1/Group2
 1080  cp ../Problem1/Group1/SampleText.txt /Desktop/Problem1/Group2
 1081  cd ..
 1082  cd Group1
 1083  ls
 1084  cp SampleText.txt /Desktop/Problem1/Group2
 1085  cp SampleText.txt /Desktop/Problem1/Group2/
 1086  cp SampleText.txt home/Desktop/Problem1/Group2
 1087  cd ..
 1088  ls
 1089  pwd
 1090  cd Group1
 1091  cp SampleText.txt home/niles/Desktop/Problem1/Group2
 1092  cp SampleText.txt /home/niles/Desktop/Problem1/Group2
 1093  cd Group2
 1094  cd ..
 1095  cd Group2
 1096  ls
 1097  nano SampleText.txt 
 1098  wc Answer1.txt 
 1099  nano SampleText.txt 
 1100  cat Answer1.txt 
 1101  rm SampleText.txt 
 1102  nano Answer1.txt 
 1103  wc Answer1.txt 
 1104  nano Answer1.txt 
 1105  wc Answer1.txt 
 1106  cat Answer1.txt 
 1107  shutdown now
 1108  cd Desktop/Problem1/Group2
 1109  ls
 1110  nano Answer2.txt
 1111  wc Answer2.txt 
 1112  nano Answer2.txt
 1113  wc Answer2.txt 
 1114  cd ..
 1115  ls
 1116  cd Shell_Scripts/
 1117  ls
 1118  nano remove.sh
 1119  chmod +x remove.sh 
 1120  ./remove.sh
 1121  nano remove.sh 
 1122  shutdown now
 1123  cd Desktop/Problem1/Shell_Scripts/
 1124  nano harden.sh 
 1125  nano sysinfo.sh
 1126  chmod +x sysinfo.sh 
 1127  ./sysinfo.sh
 1128  nano sysinfo.sh
 1129  cd Desktop/Problem1/Shell_Scripts/
 1130  ll
 1131  nano harden.sh
 1132  chmod +x harden.sh 
 1133  ./harden.sh
 1134  sudo su
 1135  cd Desktop/Problem1/Shell_Scripts/
 1136  c
 1137  ./sysinfo.sh 
 1138  c
 1139  ./sysinfo.sh 
 1140  c
 1141  ./sysinfo.sh 
 1142  ls
 1143  rm Sysinfo_date_17-05-18_time_05-28-52.html Sysinfo_date_17-05-18_time_0 Sysinfo_date_17-05-18_time_05-42-20.html Sysinfo_date_17-05-18_time_06-06-34.html Sysinfo_date_17-05-18_time_06-08-50.html Sysinfo_date_17-05-18_time_06-11-40.html 
 1144  ls
 1145  cd Desktop/Problem1/Shell_Scripts/
 1146  c
 1147  cd
 1148  cd Desktop/
 1149  LS
 1150  ls
 1151  ll
 1152  tar -czfv INFOSYS_3858_CWyatt.tar.gz Problem1
 1153  sudo tar -czfv INFOSYS_3858_CWyatt.tar.gz Problem1
 1154  sudo tar -czvf INFOSYS_3858_CWyatt.tar.gz Problem1
 1155  sudo tar -czvf INFOSYS_3858_CWyatt.tar.gz LNI
 1156  cd Desktop/Problem1/Shell_Scripts/
 1157  c
 1158  shutdown now
 1159  cd Desktop/Problem1/Shell_Scripts/
 1160  sysdiag
 1161  pstree
 1162  c
 1163  ./sysinfo.sh 
 1164  nano sysinfo.sh 
 1165  wireshark
 1166  sudo wireshark
 1167  apt-get install maltego
 1168  sudo apt-get install maltego
 1169  sudo apt-get install Maltego
 1170  rstudio
 1171  cd Downloads/
 1172  ls
 1173  ./Cytoscape_3_6_1_unix.sh
 1174  sudo ./Cytoscape_3_6_1_unix.sh
 1175  chmod +x Cytoscape_3_6_1_unix.sh 
 1176  sudo ./Cytoscape_3_6_1_unix.sh
 1177  sudo dpkg -i install
 1178  sudo dpkg -i uname
 1179  sudo dpkg -i --install
 1180  sudo dpkg -i
 1181  dpks -i
 1182  dpkg -i
 1183  sudo dpkg -i
 1184  dpkg --help
 1185  maltego
 1186  sudo ufw enable
 1187  sudo apt-get install gufw
 1188  sudo gufw
 1189  RapidMiner
 1190  sudo apt-get install clamtk
 1191  cd Desktop/
 1192  nano maltego
 1193  ls
 1194  clamscan
 1195  clamscan wiresharkcharlestoday 
 1196  refresh
 1197  sudo clamscan wiresharkcharlestoday 
 1198  ls
 1199  clear
 1200  exit
 1201  sudo apt-get install gephi
 1202  gephi
 1203  cd Downloads/
 1204  ls
 1205  tar -gzk gephi-0.9.2-linux.tar.gz 
 1206  tar -xzf gephi-0.9.2-linux.tar.gz 
 1207  cd gephi-0.9.2/
 1208  ls
 1209  cd bin/
 1210  ls
 1211  ./gephi.exe
 1212  sh gephi
 1213  gephi
 1214  ./gephi
 1215  c
 1216  cd
 1217  c
 1218  cd Downloads/
 1219  ls
 1220  cd /Desktop/
 1221  cd ..
 1222  cd Desktop/
 1223  cd rapidminer-studio/
 1224  ls
 1225  ./RapidMiner-Studio.sh
 1226  update
 1227  sudo nano .bashrc
 1228  sudo nano apt-get update
 1229  sudo apt-get update
 1230  update
 1231  sudo apt-get install rapidminor
 1232  sudo apt-get install rapid-miner
 1233  sudo apt-get install rapid miner
 1234  sudo apt-get install rapidminer
 1235  sudo apt-get install RapidMiner
 1236  sudo apt-get install Rapid-Miner

reddit_test.py
## will hopefully scrape reddit for information

#! usr/bin/env python3

import praw
import pandas as pd
import datetime as dt

def reddit_osint():
    ##connect to reddit and store in variable 'reddit'
    reddit = praw.Reddit(client_id='e3Vgbf0tP4TLGg',
                         client_secret='_Sy6Ybkvmgp94LekrcvT1U0EBwg',
                         user_agent='infosec project',
                         username='dtlz89',
                         password='sniperdude1499')

    ##accessing subreddit and storing it in 'subreddit' variable
    subreddit = reddit.subreddit('aww')

    ##creates a list of the top 50 posts in the subreddit
    top_subreddit = subreddit.top(limit=50)

    ##specifies which parts of post to capture
    topics_dict = { "title":[], \
                    "score":[], \
                    "id":[], "url":[], \
                    "comms_num": [], \
                    "body":[]}

    ##pulls information from each post
    for submission in top_subreddit:
        topics_dict["title"].append(submission.title)
        topics_dict["score"].append(submission.score)
        topics_dict["id"].append(submission.id)
        topics_dict["url"].append(submission.url)
        topics_dict["comms_num"].append(submission.num_comments)
        topics_dict["body"].append(submission.selftext)

    ##fixing date formatting
    def get_date(created):
         return dt.datetime.fromtimestamp(created)

    _timestamp = topics_data["created"].apply(get_date)

    topics_data = topics_data.assign(timestamp = _timestamp)

    ##pandas makes this look nicer

    ##pandas makes this look nicer
    topics_data = pd.DataFrame(topics_dict)

    ##creating csv
    topics_data.to_csv('redditdata.csv')


gui.py

#just tryin make uh good good ui

#Known errors:
#Twitter function runs on startup

#first test
import tkinter as tk
from tkinter import *
#from tkinter.ttk import *
from tkinter import messagebox
from reddit import reddit_osint
import subprocess as sub
import tkinter.scrolledtext as tkst
import traceback
import sys
from twitter import twitter
from tkinter.font import Font
import csv


window = Tk()

window.title("This will be the OSINT dashboard")

text_input1=Entry(window, font=('Times', 20))
text_input2=Entry(window, font=('Times', 20))
text_input3=Entry(window, font=('Times', 20))
text_input4=Entry(window, font=('Times', 20))
text_input5=Entry(window, font=('Times', 20))
text_input6=Entry(window, font=('Times', 20))
text_input7=Entry(window, font=('Times', 20))
text_input8=Entry(window, font=('Times', 20))


#eventually the functions will run the osint
def clicked_face():
    facebook=Toplevel(bg='#3b5998')
    facebook.title("FACEBOOK OSINT")
    facelbl=Label(facebook, text="osinting the facebook", bg='#3b5998', font=('Times', 20), height=10, width=25)
    facelbl.grid(row=0, column=0, columnspan=3)

def clear_text():
    window.text_box.configure(state="normal")
    window.text_box.delete(1.0, END)
    window.text_box.configure(state="disabled")

#change the label when hovering over a button for more info
def redd_label(event):
    lbl.configure(text="Search reddit", font=('Times', 12))
def twit_label(event):
    lbl.configure(text="Search twitter", font=('Times', 12))
def face_label(event):
    lbl.configure(text="Search facebook", font=('Times', 12))
def revert_label(event):
    lbl.configure(text="Select the platform to search", font=('Times', 12))

#create entry widgets for user input
def user_input():
    def clear_twit():
        label_input1.grid_forget()
        label_input2.grid_forget()
        label_input3.grid_forget()
        label_input4.grid_forget()
        label_input5.grid_forget()
        label_input6.grid_forget()
        label_input7.grid_forget()
        text_input1.grid_forget()
        text_input2.grid_forget()
        text_input3.grid_forget()
        text_input4.grid_forget()
        text_input5.grid_forget()
        text_input6.grid_forget()
        text_input7.grid_forget()
        submit_btn.grid_forget()
        clear_twit_btn.grid_forget()

    label_input1=Label(window, text="Enter consumer key", font=('Times', 12))
    label_input1.grid()
    text_input1.grid()
    label_input2=Label(window, text="Enter consumer secret:", font=('Times', 12))
    label_input2.grid()
    text_input2.grid()
    label_input3=Label(window, text="Enter access token:", font=('Times', 12))
    label_input3.grid()
    text_input3.grid()
    label_input4=Label(window, text="Enter access token secret: ", font=('Times', 12))
    label_input4.grid()
    text_input4.grid()
    label_input5=Label(window, text="Enter latitude:", font=('Times', 12))
    label_input5.grid()
    text_input5.grid()
    label_input6=Label(window, text="Enter longitude:", font=('Times', 12))
    label_input6.grid()
    text_input6.grid()
    label_input7=Label(window, text="Enter radius:", font=('Times', 12))
    label_input7.grid()
    text_input7.grid()
    submit_btn=Button(window, text="Submit", font=('Times', 12), command=submit_twit)
    submit_btn.grid()
    clear_twit_btn = Button(window, text="Clear", font=('Times', 12), command=clear_twit)
    clear_twit_btn.grid()

def submit_twit():
    cK=text_input1.get()
    cS=text_input2.get()
    aT=text_input3.get()
    aS=text_input4.get()
    lat=text_input5.get()
    long=text_input6.get()
    rad=text_input7.get()
    twitter(cK, cS, aT, aS, lat, long, rad)
   #with open("twitterdata.csv", newline = "") as file:
       #reader = csv.reader(file)

       # r and c tell us where to grid the labels
      #r = 0
       #for col in reader:
          #c = 0
          #for row in col:
             # i've added some styling
             #label = Label(window, width = 10, height = 2, \
                                   #text = row, relief = RIDGE)
             #label.grid(row = r, column = c)
             #c += 1
          #r += 1

def redd_input():
    def clear_redd():
        redd_lbl.grid_forget()
        text_input8.grid_forget()
        redd_sub_btn.grid_forget()
        clear_redd_btn.grid_forget()
        #label.grid_forget()
    redd_lbl=Label(window, text="Enter the subreddit to scrape:", font=('Times', 12))
    redd_lbl.grid()
    text_input8.grid()
    redd_sub_btn=Button(window, text="Submit", font=('Times', 12), command=submit_redd)
    redd_sub_btn.grid()
    clear_redd_btn = Button(window, text="Clear", font=('Times', 12), command=clear_redd)
    clear_redd_btn.grid()

def submit_redd():
    subr=text_input8.get()
    reddit_osint(subr)



###################################
# Trying to output to GUI instead #
# of the console                  #
###################################
#functions to redirect console output to GUI
#class IORedirector(object):
#   '''A general class for redirecting I/O to this Text widget.'''
#   def __init__(window,text_area):
#      window.text_area = text_area

#class StdoutRedirector(IORedirector):
#   '''A class for redirecting stdout to this Text widget.'''
#   def write(window,message):
#      window.text_area.insert("insert", message)
#      window.text_area.config(state = "normal")
#      window.text_area.config(state = "disabled")
#   def flush(window):
#      pass


#display the output
#window.text_box = tkst.ScrolledText(window, wrap='word', font=(20))
#window.text_box.grid(row=4, column=0, columnspan=3, rowspan=1)
#sys.stdout = StdoutRedirector(window.text_box)
#sys.stderr = StdoutRedirector(window.text_box)
#sys.last_traceback = StdoutRedirector(window.text_box)


#formatting gui window
lbl=Label(window, text="Select the platform to search", font=('Times', 12))
lbl.grid(row=0, column=0, columnspan=3)
padlbl=Label(window, height=1)
padlbl.grid(row=3)
padlbl=Label(window, height=1)
padlbl.grid(row=5)

#btn stuff
#first reddit
reddit_btn = Button(window, text="Reddit OSINT", bg='#FF4500', font=('Times', 12), height=5, width=25, command=redd_input)
reddit_btn.grid(row=2, column=0)
reddit_btn.bind("<Enter>", redd_label)
reddit_btn.bind("<Leave>", revert_label)
#then twitter
twitter_btn = Button(window, text="Twitter OSINT", bg='#0084b4', font=('Times', 12), height=5, width=25, command=user_input)
twitter_btn.grid(row=2, column=1)
twitter_btn.bind("<Enter>", twit_label)
twitter_btn.bind("<Leave>", revert_label)
#and facebook too
facebook_btn = Button(window, text="Facebook OSINT", bg='#3b5998', font=('Times', 12), height=5, width=25, command=clicked_face)
facebook_btn.grid(row=2, column=2)
facebook_btn.bind("<Enter>", face_label)
facebook_btn.bind("<Leave>", revert_label)
#clear text button
#clear_text=Button(window, text="Clear", command=clear_text)
#clear_text.grid(row=6, column=2)


window.mainloop()


twitter.py
import json
import tweepy
import pandas as pd
from tweepy import OAuthHandler

def twitter(consumer_key, consumer_secret, access_token, access_secret, lat, long, r):

	try :

		#consumer_key = input("Enter consumer key: ")
		#consumer_key = text_input.get()
		#consumer_secret = input("Enter consumer secret: ")
		#access_token = input("Enter access token: ")
		#access_secret = input("Enter access token secret: ")

		#connect with twitter api
		auth = OAuthHandler(consumer_key, consumer_secret)
		auth.set_access_token(access_token, access_secret)

		api = tweepy.API(auth)
		user = api.get_user('2018Osint')

		#print ("Processing...")

	except tweepy.TweepError:
		raise SystemExit("Error. Failed to connect to Twitter.")

	#set up csv file format
	format = { "username":[], \
 		   "tweet":[], \
		   "time":[]}

	#lat = input("Enter latitude: ")
	#long = input("Enter longitude: ")
	#r = input("Enter search radius in km: ")
	geo = "{},{},{}km".format(lat, long, r)

	geo_tweets = [status for status in tweepy.Cursor(api.search, geocode=geo).items(200)]
	for item in geo_tweets:
		format["username"].append(item.user.name)
#		if 'retweeted_status' in dir(item):
#			tweet=item.retweeted_status.full_text
#		else:
#			tweet=item.full_text
		format["tweet"].append(item.text)
		format["time"].append(item.created_at)
		#Can print other information and format or write to file as needed

	format = pd.DataFrame(format)
	format.to_csv('twitterdata.csv')

#twitter()
#added for testing


reddit.py
## will hopefully scrape reddit for information

#! usr/bin/env python3
import praw
import pandas as pd

def reddit_osint(userinput):
    ##connect to reddit and store in variable 'reddit'
    reddit = praw.Reddit(client_id='e3Vgbf0tP4TLGg', \
                         client_secret='_Sy6Ybkvmgp94LekrcvT1U0EBwg', \
                         user_agent='infosec project', \
                         username='dtlz89', \
                         password='sniperdude1499')

    ##accessing subreddit entered by user and storing it in 'subreddit'
    #userinput = input('Enter the subreddit to scrape:')
    subreddit = reddit.subreddit(userinput)

    ##creates a list of the top posts in the subreddit
    top_subreddit = subreddit.top(limit=50)

    ##specifies which parts of post to capture
    topics_dict = { "id":[], \
                    "title":[], \
                    "body":[]}

    ##checks information from each post against word or phrase in 'wordcheck'
    for submission in top_subreddit:
        wordcheck = [".onion"]
        if any(x in (submission.selftext or submission.title) for x in wordcheck):
            topics_dict["title"].append(submission.title)
            topics_dict["id"].append(submission.id)
            topics_dict["body"].append(submission.selftext)

    print("Scraping top 200 posts in r/" + userinput + " for: " + " ".join(map(str, wordcheck)))

    ##pandas makes this look nicer 
    topics_data = pd.DataFrame(topics_dict)

    ##creating csv and creating new filename if one already exists
    topics_data.to_csv("redditdata-" + userinput + ".csv")
    print('Created CSV file')

#reddit_osint()
